pcp/versions/next/unauthorized.md

```diff
@@ -0,0 +1,50 @@
+Unauthorized
+===
+
+When a broker declines to deliver a message because it fails authorization,
+it must send an *unauthorized* message back to the original message sender.
+
+A clinet may send a message to multiple recipients while it's only authorized
+to send that message to some of them:
+
+```
+    client A                    broker B                multiple clients
+       |                           |                           |
+       |         1 message         |                           |
+       |-------------------------->| 2                         |
+       |                           |                           |
+       |      3 unauthorized       |                           |
+       |<--------------------------|                           |
+       |                           |                           |
+       |                           |         4 message         |
+       |                           |-------------------------->|
+       |                           |                           |
+```
+
+Client A sends a message to possibly multiple clients through broker B (1).
+B processes (2) the message, checks its authorization, which fails for at
+least some of the clients A addressed the message to. As a result B sends
+the unauthorized message (3) to A, and at the same time it delivers the
+message (4) to the clients to which A is authorized to send the message
+in question.
+
+unauthorized messages must have the envelope *message_type* equal to
+`http://puppetlabs.com/unauthorized`.
+
+The data content is described by the following JSON schema:
+
+```
+{
+    "properties" : {
+        "id" : { "type" : "string" },
```

Comment:
@richardc should the broker provide the list of recipients for which the message won't be delivered?
Also, should we directly rely on "in_reply_to" instead of having "id"?